futures-util/src/future/join_all.rs

```diff
@@ -53,30 +61,37 @@ where
 /// # Examples
 ///
 /// ```
-/// use futures_util::future::{FutureExt, join_all, ready};
+/// #![feature(async_await, await_macro, futures_api)]
```

Comment:
This isn't new to your PR, but this change adds yet another reason to point from `join_all` to `FuturesUnordered`. After this PR, it won't be possible to add new futures to a `JoinAll`. Additionally, the fact that all futures must all be scanned even when some have completed gives another reason to prefer `FuturesUnordered`.

Can you add a comment to this doc comment about how users with large numbers of futures or who need to dynamically `push` to the collection should use `FuturesUnordered` instead?

Reply:
Yep, I see this mostly as a convenience utility when you have a handful of futures from the same `async fn` and want to simply `await!` them all (that could mostly be covered by a macro version that avoids the heap allocation, but there is still the smaller usecase of an unknown length handful of futures that this could be used for). Definitely makes sense to push users towards `FuturesUnordered` pretty aggressively.

Reply:
Cool-- sgtm!